packages/common/src/mission.rs

```diff
@@ -0,0 +1,74 @@
+use crate::{ado_base::query_get, encode_binary, error::ContractError};
+use cosmwasm_std::{Addr, Api, QuerierWrapper};
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, JsonSchema)]
+pub struct AndrAddress {
+    /// Can be either an address or identifier of an ADO in a mission.
+    pub identifier: String,
+}
+
+impl AndrAddress {
+    /// Gets the address from the given identifier by attempting to validate it into an [`Addr`] and
+    /// then querying the mission contract if it fails.
+    pub fn get_address_from_mission(
```

Comment:
I think this naming may be slightly misleading. My first impression is that this will purely work with a namespaced address rather than a contract/wallet address. Not a major issue but definitely something to consider.

Reply:
Good point, I'll try to come up with a better name